File: PlusOne.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{    
public:
    vector<int> plusOne(vector<int> &digits)
    {
        int add = 1;
        for (int i = digits.size() - 1; i >= 0; i--)
        {
            digits[i] += add;
            if (digits[i] > 9)
            {
                add = 1;
                digits[i] = digits[i] % 10;
            }
            else
            {
                add = 0;
                break;
            }
        }
        if (add == 1)
            digits.insert(digits.begin(), 1);
        return digits;
    }
};

int main()
{
    
    return 0;
}

File: rust_solution/src/plus_one.rs
pub struct Solution {}

impl Solution {
    pub fn plus_one(digits: Vec<i32>) -> Vec<i32> {
        let mut result = Vec::new();
        let mut next = 0;
        let mut digits = digits;
        let mut d = digits.pop().unwrap() + 1;
        while digits.len() >= 0 {
            if d >= 10 {
                next = 1;
                d = d % 10;
            } else {
                next = 0;
            }
            result.push(d);
            if digits.len() == 0 {
                break;
            }
            d = digits.pop().unwrap() + next;
        }

        if next > 0 {
            result.push(next);
        }
        result.reverse();
        result
    }
}